circleguard/loadable.py

```diff
@@ -706,3 +706,317 @@ def load(self, loader, cache):
+
+        Parameters
+        ----------
+        timestamps: list(int)
+            The timestamps to interpolate to.
+
+        Notes
+        -----
+        This method is in-place.
+        """
+        prev_err = np.seterr(all="ignore")
+
+        txyks = []
+
+        self.as_list_with_timestamps()
+
+        i = 1
+        t_end = self.txyk[-1][0]
+
+        for t in timestamps:
+            if t > t_end:
+                break
+
+            while self.txyk[i][0] < t:
+                i += 1
+
+            r = (t - self.txyk[i - 1][0]) / (self.txyk[i][0] - self.txyk[i - 1][0])
+
+            if np.isnan(r) or np.isinf(r):
+                continue
+
+            x = r * self.txyk[i][1] + (1 - r) * self.txyk[i - 1][1]
+            y = r * self.txyk[i][2] + (1 - r) * self.txyk[i - 1][2]
+            k = self.txyk[i - 1]
+
+            txyks += [[t, x, y, k]]
+
+        self.txyk = txyks
+
+        np.seterr(**prev_err)
+
+        return self
+
+    def shift(self, dx, dy, dt):
+        """
+        Shift the data in this :class:`~.ReplayModified` by the specified offsets.
+
+        Parameters
+        ----------
+        dx: float
+            The offset for the x coordinate.
+        dy: float
+            The offset for the y coordinate.
+        dt: int
+            The number of ticks the timestamps should be shifted.
+
+        Notes
+        -----
+        This method is in-place.
+        """
+        txyks = []
+
+        self.as_list_with_timestamps()
+
+        for t, x, y, k in self.txyk:
+            txyks += [[t + dt, x + dx, y + dy, k]]
+
+        self.txyk = txyks
+
+        return self
+
+    def as_list_with_timestamps(self):
+        """
+        Gets this replay's play data as a list of tuples of absolute time,
+        x, y, and pressed keys for each event in the data.
+        Saves the data which is returned on consequent calls.
+        This data reflects changes made by other methods.
+        
+        Returns
+        -------
+        list[tuple(int, float, float, something)]
+            A list of tuples of (t, x, y, keys) for each event
+            in the replay data.
+        """
+        if not self.txyk:
+            self.txyk = Replay.as_list_with_timestamps(self)
+
+        return self.txyk
+
+    @staticmethod
+    def copy(replay):
+        """
+        Copies the :class:`~.Replay`
+
+        Parameters
+        ----------
+        replay: :class:`~.Replay`
+            The replay to copy.
+        
+        Returns
+        -------
+        :class:`~.ReplayModified`
+            The copy as a :class:`~.ReplayModified`.
+        """
+        
+        r = ReplayModified(replay.timestamp, replay.map_id, replay.username,
+                      replay.user_id, replay.mods, replay.replay_id, replay.replay_data, replay.weight)
+        r.txyk = replay.as_list_with_timestamps()
+        return r
+
+    @staticmethod
+    def filter_single(replay):
```

Comment:
I think that only `filter_single` is useful outside of comparison so that could be moved to `utils` or `Replay` but the others should be kept close to the comparison